app/viam_client_test.go

```diff
@@ -0,0 +1,123 @@
+package app
+
+import (
+	"context"
+	"testing"
+
+	"github.com/viamrobotics/webrtc/v3"
+	"go.viam.com/utils"
+	"go.viam.com/utils/rpc"
+	"google.golang.org/grpc"
+
+	"go.viam.com/rdk/logging"
+)
+
+var (
+	logger       = logging.NewLogger("test")
+	defaultURL   = "https://app.viam.com"
+	testAPIKey   = "abcdefghijklmnopqrstuv0123456789"
+	testAPIKeyID = "abcd0123-ef45-gh67-ij89-klmnopqr01234567"
+)
+
+type MockConn struct{}
+
+func (m *MockConn) Invoke(ctx context.Context, method string, args, reply any, opts ...grpc.CallOption) error {
+	return nil
+}
+
+func (m *MockConn) NewStream(
+	ctx context.Context,
+	desc *grpc.StreamDesc,
+	method string,
+	opts ...grpc.CallOption,
+) (grpc.ClientStream, error) {
+	return nil, nil
+}
+func (m *MockConn) PeerConn() *webrtc.PeerConnection { return nil }
+func (m *MockConn) Close() error                     { return nil }
+func mockDialDirectGRPC(
+	ctx context.Context,
+	address string,
+	logger utils.ZapCompatibleLogger,
+	opts ...rpc.DialOption,
+) (rpc.ClientConn, error) {
+	return &MockConn{}, nil
+}
+
+func TestCreateViamClientWithOptions(t *testing.T) {
+	urlTests := []struct {
+		name      string
+		baseURL   string
+		entity    string
+		payload   string
+		expectErr bool
+	}{
+		{"Default URL", defaultURL, testAPIKeyID, testAPIKey, false},
+		{"Default URL", defaultURL, "", "", true},
+		{"Default URL", defaultURL, "", testAPIKey, true},
+		{"Default URL", defaultURL, testAPIKeyID, "", true},
+		{"Empty URL", "", testAPIKeyID, testAPIKey, false},
```

Comment:
Try wil `nil` and with `""`

Reply:
I tested this separately but didn't actually add it as a test. Thanks for catching!